images/router/haproxy/conf/haproxy-config.template

```diff
@@ -348,6 +364,22 @@ frontend fe_sni
   http-request set-header X-SSL-Client-DER       %{+Q}[ssl_c_der,base64]
     {{- end }}
 
+    {{- range $idx, $http_request_header := .HTTPRequestHeaders }}
```

Comment:
Again, there are so many headers we explicitly set here, why would we allow the user to change those?

Reply:
This section is set by the admin via ingress controller so I think this shall be his responsibility  to set the headers correctly.

Reply:
other option is to add this code just before https://github.com/openshift/router/pull/438/files#diff-21bd64a4ad8a7927e9d59e99e09628efc884fe6d6291e50eb33c5deb3d2a79f0R354